syntax = "proto3";

package books_database;

service BooksDatabase {
    rpc Read (ReadRequest) returns (ReadResponse) {}
    rpc Write (WriteRequest) returns (WriteResponse) {}
    rpc IncrementStock (WriteRequest) returns (WriteResponse) {}
    rpc DecrementStock (WriteRequest) returns (WriteResponse) {}
}

message ReadRequest {
    string title = 1;
}

message ReadResponse {
    int32 stock = 1;
}

message WriteRequest {
    string title = 1;
    int32 quantity = 2;
}

message WriteResponse {
    bool success = 1;
}
